jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/eks/CheckEKSClusterVersionIsUpdatedRule.java

```diff
@@ -0,0 +1,114 @@
+        }
+        Optional<String> opt = Optional.ofNullable(resourceAttributes)
+                .map(resource -> checkValidation(ruleParam, resource));
+        RuleResult ruleResult = Optional.of(ruleParam).filter(param -> opt.isPresent())
+                .map(param -> buildFailureAnnotation(param, opt.get()))
+                .orElse(new RuleResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE));
+        logger.debug("========CheckForEKSClusterVersionUpdate ended=========");
+        return ruleResult;
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Checks the EKS Cluster version is updated to latest or not";
+    }
+
+    private String checkValidation(Map<String, String> ruleParam, Map<String, String> resource) {
+        String accountId = resource.get(PacmanRuleConstants.ACCOUNTID);
+        String clusterName = resource.get(PacmanRuleConstants.CLUSTER_NAME);
+        String currentClusterVersion = resource.get(PacmanRuleConstants.VERSION);
+        String latestClusterVersion = ruleParam.get(PacmanRuleConstants.LATEST_VERSION);
+
+        if (!PacmanUtils.doesAllHaveValue(accountId, clusterName, currentClusterVersion, latestClusterVersion)) {
+            logger.info(PacmanRuleConstants.MISSING_CONFIGURATION);
+            throw new InvalidInputException(PacmanRuleConstants.MISSING_CONFIGURATION);
+        }
+
+        if (compareVersions(currentClusterVersion, latestClusterVersion) < 0) {
+            return "Update your EKS cluster version for EKS Cluster: " + clusterName + ",accountId: " + accountId
+                    + ",currentClusterVersion" + currentClusterVersion + ",latestClusterVersion" + latestClusterVersion;
+        }
+        return null;
+    }
+
+    private static RuleResult buildFailureAnnotation(final Map<String, String> ruleParam, String description) {
+        LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+        List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+        Annotation annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+        annotation.put(PacmanSdkConstants.DESCRIPTION, description);
+        annotation.put(PacmanRuleConstants.SEVERITY, ruleParam.get(PacmanRuleConstants.SEVERITY));
+        annotation.put(PacmanRuleConstants.CATEGORY, ruleParam.get(PacmanRuleConstants.CATEGORY));
+        annotation.put(PacmanRuleConstants.RESOURCE_ID, ruleParam.get(PacmanRuleConstants.RESOURCE_ID));
+        issue.put(PacmanRuleConstants.VIOLATION_REASON, description);
+        issueList.add(issue);
+        annotation.put("issueDetails", issueList.toString());
+        logger.debug("========CheckForEKSClusterVersionUpdate annotation {} :=========", annotation);
+        return new RuleResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE, annotation);
+    }
+
+    private static int compareVersions(String version1, String version2) {
```

Comment:
To compare the versions we can parse the string as double(or float) and can do double.compare().

Reply:
Ahh no, When I checked, I found that it won't be able to pick 1.11 as the higher version when we compare 1.9 and 1.11